### DISCUSSION TITLE
Playbook Finds Wrong Data

### DISCUSSION THREAD

**Discussion by OmerTirosh** [Security-Onion-Solutions/securityonion#2261] [Unsupported Versions] [answered]
Hi, 

I tried to write new alerts with playbook looking in subpath of field and I noticed that in all the alerts that starts or ends with '*' are incorrect. 

For example Renamed Powershell play (320) also finds event that ends with 'powershell.exe' 
 
Maybe the problam is with the security analyzer?
Can it find data based on partial data?

Thank you.

**Comment by defensivedepth** [comment]
@OmerTirosh Can you please post a bit more about what is going? The more detail, the better:

- Version of SO, what kind of install
- An example of a log and a sigma that you are trying to write but isn't working, etc.

**Reply by OmerTirosh** [reply]
@defensivedepth Can you please check it out? Do you need any more information? 

I mostly want to know if this is a bug...

**Reply by defensivedepth** [reply]
@OmerTirosh Please take a minute and review the Support Guidelines - https://github.com/Security-Onion-Solutions/securityonion/discussions/1720

`Community Support is considered best effort and there are no guarantees and no SLAs. If you need private or priority support, please consider purchasing support from Security Onion Solutions.`

I do believe this is an issue with the analyzer - we are looking into it further. Thanks!

**Comment by OmerTirosh** [comment]
I'm using SO 2.3.0 distributed mode

'Renamed Powershell' rule (from the playbook) :

```
detection:
  selection:
    Description: Windows Powershell
    Company: Microsoft Corporation
  filter:
    Image:
    -  '*\powershell.exe'
    -  '*\powershell_ise.exe'
  condition: selection and not filter 
```

The elastalert query: 
`(event.code.security: "1" AND winlog.channel.security: "Microsoft\-Windows\-Sysmon\/Operational" AND (process.pe.description.security: "Windows\ PowerShell" AND process.pe.company.security: "Microsoft\ Corporation") AND ( NOT (process.executable.security: ("*\\powershell.exe" OR "*\\powershell_ise.exe"))))`

All the results in 'process.executable' field is ending with powershell.exe when its not suppose to be, Example: "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe".

The problem seems to be with the es_security_analyzer or the es_security_search_analyzer, because if you search in a standard analyzer the results are correct.

**Comment by defensivedepth** [comment, marked as answer]
2.3.30 has been released that fixes these issues - https://blog.securityonion.net/2021/03/security-onion-2330-now-available.html
